test/gtest/ucp/test_ucp_am.cc

```diff
@@ -692,6 +692,141 @@ class test_ucp_am_nbx : public test_ucp_am_base {
     ucp_mem_h                       m_rx_memh;
 };
 
+class test_ucp_am_id : public test_ucp_am_nbx {
+public:
+    test_ucp_am_id()
+    {
+        reset_counters();
+    }
+
+protected:
+
+    void reset_counters()
+    {
+        test_ucp_am_nbx::reset_counters();
+        m_recv_counter_cb_1 = 0;
+        m_recv_counter_cb_2 = 0;
+    }
+
+    ucs_status_t set_am_id_overflow_data_handler(entity &e, unsigned am_id,
+                               ucp_am_recv_callback_t cb, void *arg,
+                               unsigned flags = 0)
+    {
+        ucp_am_handler_param_t param;
+
+        /* Initialize Active Message data handler */
+        param.field_mask = UCP_AM_HANDLER_PARAM_FIELD_ID |
+                           UCP_AM_HANDLER_PARAM_FIELD_CB |
+                           UCP_AM_HANDLER_PARAM_FIELD_ARG;
+        param.id         = am_id;
+        param.cb         = cb;
+        param.arg        = arg;
+
+        if (flags != 0) {
+            param.field_mask |= UCP_AM_HANDLER_PARAM_FIELD_FLAGS;
+            param.flags       = flags;
+        }
+
+        return ucp_worker_set_am_recv_handler(e.worker(), &param);
+    }
+
+    void test_am_id_handler(size_t size, size_t header_size = 0ul)
+    {
+        mem_buffer sbuf(size, tx_memtype());
+        sbuf.pattern_fill(SEED);
+        m_hdr.resize(header_size);
+        ucs::fill_random(m_hdr);
+        reset_counters();
+        ucp_mem_h memh = NULL;
+
+        EXPECT_EQ(UCS_OK, set_am_id_overflow_data_handler(receiver(),
+                                        0x1, am_id_overflow_data_cb_1,
+                                        this, 0));
```

Comment:
I think you don't need to pass 2 last parameters - `this` can be taken from the `set_am_id_overflow_data_handler`
And `flags` are not used, so no need to have that param at all